src/main/java/Deadline.java

```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    private String doBy;
+
+    public Deadline (String description, String deadline) {
+        super(description);
+        this.doBy = deadline;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + "[" + super.getStatusIcon() + "] " + super.getDescription() + " (by: " + this.doBy + ")";
+    }
```

Comment:
I feel like there is no need to use `super.getStatusIcon` and `super.getDescription` since your Deadline already extends from Tasks. Hence, these methods already exist within the Deadline class. Furthermore, the use of this should be removed if there is no multiple instances of it. This is a suggestion.
```suggestion
    public String toString() {
        return String.format("[D] [%s] %s (by: %s)", getStatusIcon(), getDescription(), doBy);
    }
```
The other classes that extend from Task have similar problem and can be dealt with respectively as well.